paxml/learners.py

```diff
@@ -416,6 +438,9 @@ def _adjust_var(old_var, transformed_grad, is_learnable, var_wh, var_key):
       if not is_learnable:
         return old_var
 
+      if base_layer.var_overwrite_with_gradient(var_wh):
```

Comment:
We just need to attach documentation somewhere in the repo.
`var_overwrite_with_gradient` function seems like the best place.